src/SqlAsyncEnumerable.cs

```diff
@@ -106,6 +107,8 @@ private async Task<bool> GetNextRowAsync()
                     using (SqlCommand command = SqlBindingUtilities.BuildCommand(this._attribute, this._connection))
                     {
                         await command.Connection.OpenAsync();
+                        Dictionary<TelemetryPropertyName, string> props = command.Connection.AsConnectionProps();
```

Comment:
My concern here is that since we're not sending this event until GetNextRowAsync is called it's going to behave differently than all the other ones (which immediately send the event when the convert is called). 

My assumption based on this code is that if someone made a function which used IAsyncEnumerable but then never actually called MoveNextAsync that we'd never fire this event and thus get a lower count of people using this. That doesn't seem like a super reasonable case that I can think of, but in general we still should avoid knowingly having gaps in our telemetry.

Instead it seems like we should discuss opening the connection as soon as the AsyncEnumerable is created, or passing in an already open connection (but still letting the dispose close it). That way we could send the event as soon as convert is called and still lazily get the rows from the DB. 

@lucyzhang929 @chlafreniere Thoughts here?

Reply:
I think opening the connection when SqlAsyncEnumerable is created makes sense here.

Reply:
My suggestion for that would be : 

1. Open the connection in the constructor
2. Making the connection a public readonly property
3. In the code that creates the AsyncEnumerable (`IConverter<SqlAttribute, IAsyncEnumerable<T>>.Convert`) create the enumerable, use the connection from it to send the event and then return that enumerable

This keeps the event logic all in the same place which is easier to work with.